src/scheduler/node_info.c

```diff
@@ -222,6 +225,32 @@ query_nodes(int pbs_sd, server_info *sinfo)
 		sinfo->nodes_by_NASrank[i] = ninfo;
 #endif /* localmod 049 */
 
+		if (dflt_sched) {
+			if (ninfo->partition != NULL) {
+				cur_node = cur_node->next;
+				continue;
+			}
+		} else {
+			if (ninfo->partition == NULL) {
+				cur_node = cur_node->next;
+				continue;
+			} else {
+				int i;
+				int in_partition = 0;
+				for (i=0; my_partitions[i] != NULL; i++) {
+					if (strcmp(ninfo->partition, my_partitions[i]) == 0) {
+						in_partition = 1;
+						break;
+					}
+				}
+				if (!in_partition) {
+					cur_node = cur_node->next;
+					continue;
+				}
+
+			}
+		}
```

Comment:
The scheduler style avoids to use of a continue if at all possible.  It is definitely possible here.  Move this code into a function called something like node_in_partition().  It can return 1 if the node is in the partition.  You can easily use that in an if().  Plus, you're leaking memory.  If you don't put ninfo in ninfo_arr, it will never be freed.

Reply:
Implemented. Also done similar changes for querying/filtering queues info.